pkg/grpc/marshaler.go

```diff
@@ -0,0 +1,77 @@
+package grpc
+
+import (
+	"encoding/json"
+	"io"
+	"time"
+
+	"github.com/grpc-ecosystem/grpc-gateway/v2/runtime"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/pkg/protocompat"
+	"google.golang.org/protobuf/encoding/protojson"
+	"google.golang.org/protobuf/proto"
+)
+
+// Deprecated: customMarshaler is a hack for keeping backward compatibility for duration objects.
+// TODO(ROX-25678): This should be deleted in 4.8.
+type customMarshaler struct {
+	*runtime.JSONPb
+}
+
+func (c customMarshaler) Unmarshal(data []byte, v interface{}) error {
+	unmarshalError := protojson.Unmarshal(data, v.(proto.Message))
+	if unmarshalError == nil {
+		return nil
+	}
+	suppressCVERequest, ok := v.(*v1.SuppressCVERequest)
```

Comment:
I think we should explain why `v1.SuppressCVERequest` suddenly shows up here in such a general location. Maybe extract everything related to `v1.SuppressCVERequest` into a separate function with some explaining comment? Would also make it more obvious how one would expand the hack to other proto messages, if we would ever need that in the future.